### ISSUE TITLE
WebRTC leaking despite enabling the option [8.9]

### ISSUE THREAD

**Issue by grappelli** [foxyproxy/browser-extension#162]
On chromeOS 130* using the foxyproxy chrome extension 8.9 - webrtc is leaking for me.

Tested on several sites such as https://browserleaks.com/webrtc. I tried the basic app too, in both cases I accepted the additional permissions for making changes to adjusting privacy settings. No change in webrtc leaks for either of the foxyproxy extensions nor after a reboot. Both were installed separately

OTOH, The 'WebRTC Control' extension on the chrome webstore functions perfectly in comparison and is a very small and easy to look at extension - might be helpful?

Thanks

**Comment by ericjung** [comment]
I cannot reproduce this.

Steps:

1. Configure HTTPS proxy server with authentication in FoxyProxy 8.9 on Chrome 131.0.6778.205 - MacOS
2. Click the checkbox "Limit WebRTC" in FoxyProxy Options. Allow extra permissions. MUST CLICK THE CHECKBOX AGAIN because it is still not checked after accepting permissions.
3. MUST CLICK SAVE BUTTON in FoxyProxy
4. Set FoxyProxy to use the proxy server in #1. Note: I am not using patterns
5. Visit https://browserleaks.com/webrtc and see this screenshot:

![webrtc-good](https://github.com/user-attachments/assets/af120ad5-5052-4c7a-87c5-835b22ddf2cc)

**Comment by grappelli** [comment]
I'm sorry I didn't receive any notifications  on this issue, I just happened to come across your comments now.

To provide an update regardless, when I have only foxyproxy extension installed and with the 'limit webrtc' toggled on, this is my result:

![Image](https://github.com/user-attachments/assets/4f7ec81e-6ce2-48b4-8685-ee70cf9d2ae0)

when I use google's own 'WebRTC Network Limiter' then this is what I get:

![Image](https://github.com/user-attachments/assets/a529cff7-0bea-4479-8100-616f9e48e993)

thanks

**Comment by erosman** [comment]
Are you using "Proxy by Pattern" or individual proxy (all through one)?

**Comment by grappelli** [comment]
So I just retested again for you.

I can reproduce the exact same behavior in both 'proxy by patterns', and with a single proxy defined with no exemptions. New plugin install, fresh browser, etc... all defaults.

If it helps, I'm using mullvad socks5 proxy which gives a different public IP address via their VPN server (differs from the VPN servedr's public IP). When I use Google's plugin it shuts down any webrtc detection in browser leaks website. When I use foxy proxy web rtc option, it seems to do nothing wrt browser leaks' tests and it detects that my IP isn't what it should be.

**Comment by grappelli** [comment]
![Image](https://github.com/user-attachments/assets/02e5a382-8d08-400b-999c-51218e3ef380)

FYI these are the necessary settings for Google's 'WebRTC Network Limiter' extension to achieve the desired results.

**Comment by grappelli** [comment]
Fair enough... maybe its worth getting rid of the check box for chrome users then if it doesn't work and isn't going to be fixed?

**Comment by erosman** [comment]
JavaScript runs client-side, which is user's computer.

Unless the JavaScript has a code to send the data to a remote server, or set values such as cookies that can be read later, what users see if not necessarily shown to the server.

The FoxyProxy options "Limit WebRTC" limits (reduces) the leak via net requests.
Furthermore, it was not reproducible by eric which could mean something specific to your settings.

**Comment by ericjung** [comment]
Note: I tested with HTTPS proxy server. It seems @grappelli tested with a SOCKS5 proxy server.

**Comment by grappelli** [comment]
I used guest browsing each time, ie. a completely fresh chrome profile with no other extensions installed to test foxyproxy with due diligence so we can rule out 'specific settings'.

Perhaps the part of eric's testing that was less reproducible was using chromium instead of chrome or more importantly, _the lack of a socks5 server that provides the client a different public IP than its own?_

I could provide temporary access to a mullvad wireguard config to test this, but as you said, it is likely beyond the scope of your extension anyway.

As far as javascript running locally however, I find your posture a little bizarre. Any website that runs javascript that is able to detect IP leaks via webrtc obviously has the ability and even the inclination to log this information... otherwise why would they be abusing webrtc calls in the first place?

'WebRTC Control' which i mentioned in OP. is another extension that handles webrtc leaks properly but seeing as it requires access and modification to all sites, I assume that means it is injecting JS too. I am surprised that google's extension is injecting JS however, seeing as its permissions are limited to 'Change your privacy-related settings'.

Anyway, as a user, I found the webrtc toggle in foxyproxy to be a complete waste of time. It didn't work as expected - testing showed IP leaking. So I then went and read the whole foxyproxy website and FAQ and noted the mention that it 'limits' access. Then when testing other extensions to solve my leak, I realised that foxyproxy was the outlier. Then I made this bug report. Overall, more of a time wasting toggle than a webrtc limiting toggle AFAIAC ;-)

Anyway, thanks for looking into it.

**Comment by erosman** [comment]
FoxyProxy's main function is proxying. 
"Limit WebRTC" feature was added upon request to provide a convenient method to toggle browser WebRTC settings. Inadequacies in the browser settings are are better discussed with the browser developers directly.
It is up to the users to use the option or install a dedicated extension to control WebRTC.

**Comment by grappelli** [comment]
Ok well, your toggle didn't work for me, hence the report and if you don't want to fix or remove a toggle that doesn't work - that is entirely your prerogative!

I think expecting me to get google to modify their browser when they already provide a functional extension, is a bit disingenuous however.

And honestly, it's hardly a stretch to think that a proxy extension that has a webrtc toggle, might also want to ensure that webrtc is not leaking outside the proxy when that toggle is active.

I understand your position though, thanks again!

**Comment by grappelli** [comment]
Just to add as I realise there was some confusion:

https://chromewebstore.google.com/detail/webrtc-network-limiter/npeicpdbkakmehahjeeohfdhnlpdklia

this is chrome extension by google called 'WebRTC Network Limiter' 

It seems to work effectively but it does not inject javascript to do so - its only permission are the same as foxyproxy's... to 'modifiy privacy settings'.

Therefore, it does seem that if you were inclined to fix the issue, there is a method that is within the scope of foxyproxy.

Thanks again

**Comment by erosman** [comment]
FoxyProxy's "Limit WebRTC" sets "default_public_interface_only" (not all options are suitable for proxying).
The option was discussed in length with Mozilla engineers in [Bug 1790270](https://bugzilla.mozilla.org/show_bug.cgi?id=1790270).

"WebRTC Network Limiter" also sets above but has 2 other options.
Have you tested it with "default_public_interface_only"?

**Comment by erosman** [comment]
FoxyProxy's "Limit WebRTC" sets "default_public_interface_only" (not all options are suitable for proxying).
The option was discussed at length with Mozilla engineers in [Bug 1790270](https://bugzilla.mozilla.org/show_bug.cgi?id=1790270).

"WebRTC Network Limiter" also sets above but has 2 other options.
Have you tested it with "default_public_interface_only"?

**Comment by grappelli** [comment]
The last option 'use my proxy server' was the only one that consistently passed all webrtc tests for me. The next option up 'use defualt public IP' produced mixed results.

I'm not really following your link wrg to this, it just mainly goes over my head beyond maybe that the options weren't behaving as expected.

**Comment by erosman** [comment]
I will do some more testing and get back to you.

**Comment by erosman** [comment]
I did some tests. I will update the WebRTC setting for v8.11.
